### PR TITLE
Fix SSO; fix JS errors on login page; fix method name error

### PR DESCRIPTION
**Pull request by joshcanhelp**
Fixing a few issues found during testing. See comments in code for details.

### CODE REVIEW

File: auth0.module [hunk 1]
```diff
@@ -13,23 +13,14 @@ define('AUTH0_DEFAULT_USERNAME_CLAIM', 'nickname');
  * Replace a form with the lock widget.
  */
 function auth0_theme() {
-  return array(
-    'auth0_login' => array(
+  return [
```
Comment (joshcanhelp) [inline auth0.module]: All variables except `loginCSS` are now loaded in the external JS file.

File: js/auth0.lock.js [hunk 2]
```diff
@@ -13,7 +13,7 @@
         return;
       }
 
-      var lock_options = auth0.lockExtraSettings|raw;
+      var lock_options = JSON.parse( auth0.lockExtraSettings ) || {};
```
Comment (joshcanhelp) [inline js/auth0.lock.js]: @rob3000 - I'm not sure if I'm doing this right or not but `raw` was throwing an error.

File: js/auth0.lock.js [hunk 3]
```diff
@@ -30,7 +30,7 @@
         }
       }
 
-      var lock = new Auth0Lock(auth0.clientID, auth0.domain, lock_options);
+      var lock = new Auth0Lock(auth0.clientId, auth0.domain, lock_options);
```
Comment (joshcanhelp) [inline js/auth0.lock.js]: Typo

File: src/Controller/AuthController.php [hunk 4]
```diff
@@ -40,8 +40,6 @@
 use Auth0\SDK\JWTVerifier;
 use Auth0\SDK\Auth0;
 use Auth0\SDK\API\Authentication;
-use Auth0\SDK\API\Management;
```
Comment (joshcanhelp) [inline src/Controller/AuthController.php]: Unused

File: src/Controller/AuthController.php [hunk 5]
```diff
@@ -254,7 +252,7 @@ public function login(Request $request) {
     }
 
     // If supporting SSO, redirect to the hosted login page for authorization.
-    if ($this->redirectForSso === TRUE) {
+    if ($this->redirectForSso) {
```
Comment (joshcanhelp) [inline src/Controller/AuthController.php]: The value on the form is `"1"` so this explicit check for `TRUE` was failing.

File: src/Controller/AuthController.php [hunk 6]
```diff
@@ -275,7 +273,7 @@ public function login(Request $request) {
             'configurationBaseUrl' => 'https://cdn.auth0.com',
             'showSignup' => $this->config->get('auth0_allow_signup'),
             'callbackURL' => "$base_root/auth0/callback",
-            'state' => $this->helper->getNonce($returnTo),
+            'state' => $this->getNonce($returnTo),
```
Comment (joshcanhelp) [inline src/Controller/AuthController.php]: `getNonce()` is in this class.

**Comment by joshcanhelp** [comment]
@rob3000 - Can you take a look at this and make sure everything looks good? Tagged you on a specific change that I wasn't sure works or not. Ran `vendor/bin/phpcs modules/auth0/src --standard=Drupal` and only saw the one error for `verify_email`

File: src/Controller/AuthController.php [hunk 7]
```diff
@@ -275,9 +273,10 @@ public function login(Request $request) {
...
             'scopes' => AUTH0_DEFAULT_SCOPES,
             'offlineAccess' => $this->offlineAccess,
+            'formTitle' => $this->config->get('auth0_form_title'),
```
Comment (joshcanhelp) [inline src/Controller/AuthController.php]: Adding JS var for form title (was not changing the login form title)

File: js/auth0.lock.js [hunk 8]
```diff
@@ -23,14 +23,16 @@
       lock_options.auth.params = lock_options.auth.params || {};
       lock_options.auth.params.scope = lock_options.auth.params.scope || auth0.scopes;
       lock_options.auth.params.state = auth0.state;
+      lock_options.languageDictionary = lock_options.languageDictionary || {};
+      lock_options.languageDictionary.title = lock_options.languageDictionary.title || auth0.formTitle;
```
Comment (joshcanhelp) [inline js/auth0.lock.js]: Add form title customization

File: js/auth0.lock.js (hunk 2 above)
Comment (rob3000) [reply js/auth0.lock.js]: Yep thats good, raw was most likely left over from the twig template.

File: templates/auth0-login.html.twig [hunk 9]
```diff
@@ -1,31 +1,4 @@
 <div id="auth0-login-form" style=" min-height: 440px;"></div>
-
```
[the remark this answers is not in the document]
Comment (rob3000) [reply templates/auth0-login.html.twig]: We can look at loading the CSS in the same way as the remote JS too, but that can be done on a seperate PR.

**Pull request merged**